.gitignore

```diff
@@ -116,4 +116,7 @@ $RECYCLE.BIN/
 
 # Others
 data/
-.next/
\ No newline at end of file
+.next/
+
+# generate lock files
+package-lock.json
```

Comment:
`package-lock.json` should not be ignored. And we should both either use either `yarn` or `npm` only not both. As this repo already has `yarn.lock` file so you should continue with it.